sdk/typescript/src/types/common.ts

```diff
@@ -57,7 +57,7 @@ export type ObjectOwner = Infer<typeof ObjectOwner>;
 
 // TODO: Figure out if we actually should have validaton on this:
 export const SuiJsonValue = unknown();
-export type SuiJsonValue = boolean | number | string | Array<SuiJsonValue>;
+export type SuiJsonValue = boolean | number | string | PureArg | Array<SuiJsonValue>;
```

Comment:
MoveCallTransaction interface has this property `arguments: SuiJsonValue[]` so I added PureArg as SuiJsonValue but I am not sure if this is an acceptable way to do it

Reply:
`PureArg` should not be part of `SuiJsonValue` as it does not work with the RpcTxnBuilder. I see two ways of fixing this:

1. change `arguments: SuiJsonValue[]` to `arguments: (SuiJsonValue | PureArg)[]`. To make this compatible with the RpcTxnDataSerializer, we either throw an error if `PureArg` is passed as an argument in RpcTxnDataSerializer.ts 

2. Or we can update the rust code https://github.com/MystenLabs/sui/blob/d5754e7aac4a5d3d2e96e87a43a7379a219c3fda/crates/sui-json-rpc/src/transaction_builder_api.rs#L222 to include PureArg in SuiJsonValue


cc @oxade